### PR TITLE
Integrated content

### PR DESCRIPTION
**Pull request by AnnJohn**

### CODE REVIEW

**Comment by jordanhudgens** [comment]
@AnnJohn I've added the metadata.

File: README.md [hunk 1]
```diff
@@ -1,29 +1,78 @@
 # Intro to REST
 
-## Objectives
+## What is REST?
 
-1. Define the RESTful URL conventions
-2. Identify seven RESTful URLs and HTTP verbs
-3. Identify seven RESTful actions
+REST stands for REpresentational State Transfer, essentially REST is a standardized design architecture that enables applications to communicate and manage data flow. REST was originally proposed in 2000 by Roy Thomas Fielding in his PhD dissertation as an architectural style for connecting the Internet. At that time the Internet was growing fast, however there was a distinct lack of organization when it came to web applications communicating with each other. RESTful principles helped to bridge the gap between applications and aided in laying the foundation for the Internet ecosystem that exists today.
```
Comment (jmburges) [inline README.md]: This section is super super jargon-y. I would rephrase this to be:

"In 2000, Roy Thomas Fielding was frustrated by the hap-hazard different ways web applications were using the HTTP standard. Specifically he was frustrated with how URLs and their corresponding HTTP Verbs were used differently for every single application. So, in his PhD dissertation he came up with REST or REpresentational State Transfer as a standard way web apps should structure their URLs. It also suggested a few other things, but we focus mostly on how it changed URLs. Fielding, also noticed the rise in web applications communicating with each other. Using this standard way of forming URLs to access resources, Fielding hoped that inter-application communication would get much easier."

File: README.md [hunk 2]
```diff
@@ -1,29 +1,78 @@
...
-put it all together and we get predictable guessable urls that describe the things they are representing. we call these things a resource.
+5. Lastly the browser receives the server information and gives the user feedback, in this case it shows you a message saying that your newsletter was successfully posted.
 
-/posts vs /pictures
 
-the 7 restful routes 7 restful actions
+## RESTful Conventions in Rails
 
-REST is a concept - not a technology. for right now, just think of it as standard URLS and HTTP verbs that describe what they are trying to do
+Rails has RESTful principles built into its core, so whether you are utilizing the built in view rending system or using the application purely as API you will have the tools necessary to follow standardized routing procedures.
```
Comment (jmburges) [inline README.md]: rendering

Comment (jmburges) [inline README.md]: "as an API"

File: README.md [hunk 3]
```diff
@@ -1,29 +1,78 @@
...
 
-diagram that shows the mirroring of CRUD, to ROutes to controller Actions
+If you were to build out a feature in Rails that entails all of the potential RESTful routes, you would have seven different RESTful URLs, HTTP verbs (more on those later), and actions. As an example, imagine that you have a Newsletter module in your application, if you run ```rake routes``` in the terminal you will receive the following output:
```
Comment (jmburges) [inline README.md]: I would somewhat reframe this starting at something they know (basic CRUD actions) to something new (rake routes and such).

So it would be like
1. If we were trying to create a Newsletter feature (be careful of the word "module" don't want students to think it's a Ruby Module) we would have four main actions. Create, Read, Update and Delete. You know these as the CRUD actions.
2. In addition to the standard CRUD actions, we need an index page to show all of the newsletters
3. And we need a form for creating new newsletters, and a form for updating existing newsletters.
4. All those together and we have seven routes. The GET routes are all routes that are usually render some erb content. The POST and PUT are receivers of forms, and the DELETE is a new type of verb.
5. table of routes as the map from url/verb to english description
6. Thankfully, rails maps these specific things to specific methods or "actions" as they are called in rails. If we had a controller called `NewsletterController` we would define these seven methods and rails automatically will call them based on the correct route. (code snippet of the seven methods, with a comment that is the english description)

File: README.md [hunk 4]
```diff
@@ -1,29 +1,78 @@
...
+```
+newsletters     GET     /newsletters(.:format)          newsletters#index
+                POST    /newsletters(.:format)          newsletters#create
+new_newsletter  GET     /newsletters/new(.:format)      newsletters#new
+edit_newsletter GET     /newsletters/:id/edit(.:format) newsletters#edit
+newsletter      GET     /newsletters/:id(.:format)      newsletters#show
+                PATCH   /newsletters/:id(.:format)      newsletters#update
+                PUT     /newsletters/:id(.:format)      newsletters#update
+                DELETE  /newsletters/:id(.:format)      newsletters#destroy
+```
```
Comment (jmburges) [inline README.md:52]: The issue is that they are going to type `rake routes` in this folder and expect that output. Also there is extra stuff there (that left column and the .format stuff) that will confuse them

File: README.md [hunk 5]
```diff
@@ -1,29 +1,78 @@
...
+
+2. The router in the application processes the request and 'routes' the request data to the proper controller action
+
+3. The controller action either performs a task, such as creating, updating, or deleting a record in the database, or it runs a database query and renders a view to the client
+
+## Definition of HTTP Verbs
+
+So what do those 'GET', 'POST', et al items represent? Those are HTTP verbs that each give the HTTP request unique behavior, below is an explanation on each verb:
+
+* **GET** - The GET method means retrieve whatever information (in the form of an entity) is identified by the Request-URI. This means if you go to /posts you will get all of the posts that the application wants your application to have.
```
Comment (jmburges) [inline README.md]: The first sentence doesn't make sense to me. I think what you mean is `The GET method means "retrieve whatever information is identified by the URI"`. The quotes help maybe?

File: README.md [hunk 6]
```diff
@@ -1,29 +1,78 @@
...
+
+* **POST** - The POST method is used to request that the origin server accept the entity enclosed in the request as a new subordinate of the resource identified by the Request-URI in the Request-Line. This means that the server needs to accept the data sent over with the request.
```
Comment (jmburges) [inline README.md]: This sentence is super jargon-y: Maybe something like `The POST method is used to send data enclosed in the request to the server. The server is expected to use this data to create some new resource`

File: README.md [hunk 7]
```diff
@@ -1,29 +1,78 @@
...
+
+* **PATCH/PUT** - The PUT method requests that the enclosed entity be stored under the supplied Request-URI. If the Request-URI refers to an already existing resource, the enclosed entity SHOULD be considered as a modified version of the one residing on the origin server. This means that the server needs to accept data and that it is understood that the data is going to modify a pre-existing element in the database.
```
Comment (jmburges) [inline README.md]: Jargon-y. "enclosed entity" is alien to most students

File: README.md [hunk 8]
```diff
@@ -1,29 +1,78 @@
...
+
+* **DELETE** - The DELETE method requests that the origin server delete the resource identified by the Request-URI. This means… it deletes the record, it's nice and explicit.
```
Comment (jmburges) [inline README.md]: Jargon-y. Also they don't understand "origin server". "server" is probably fine

File: README.md [hunk 9]
```diff
@@ -1,29 +1,78 @@
...
+
+
+## Summary
+
+Below are a few keys to remember when thinking about REST:
+
+* REST is an architectural design pattern, not a framework or code in itself.
```
Comment (jmburges) [inline README.md]: What does that mean in the real world? Other frameworks, implement the REST pattern. Rails is not unique in this manner

File: README.md [hunk 10]
```diff
@@ -1,29 +1,78 @@
...
+                PATCH   /newsletters/:id(.:format)      newsletters#update
+                PUT     /newsletters/:id(.:format)      newsletters#update
+                DELETE  /newsletters/:id(.:format)      newsletters#destroy
+```
+
+Rails does a great job of integrating RESTful routes into their system, if you can understand routes in Rails you can understand REST in general. Going through the output above you should be able to see that all of the potential CRUD actions are all there, from querying all of the records to deleting a single item from the database, and all of the actions are wired up using RESTful routing nomenclature.
+
+Here is a diagram that shows how how the views, controller actions, routes, and HTTP verbs are all mapped together
+
+![REST Diagram](http://reif.io/lib/flatiron/rest_diagram.png)
```
Comment (jmburges) [inline README.md:58]: This diagram is good except I would combine the method and the path into just one box each.

**Comment by jmburges** [comment]
back at you

**Comment by jmburges** [comment]
I would also add to this the "mechanics" of restful routes in rails. Show them that they could set up each of these routes individually with seven different lines...and then reveal the magic of the  `resource` routes

**Comment by jordanhudgens** [comment]
@jmburges I made a lot of changes to this one and incorporated each of the items you brought up, the only one I wasn't sure on was the diagram change. Did you want the path combined with the controller method or the HTTP verb?

**Comment by jmburges** [comment]
hey @jordanhudgens do you have the original of the diagram? In PSD or sketch or w/e? I'm happy to make the change
